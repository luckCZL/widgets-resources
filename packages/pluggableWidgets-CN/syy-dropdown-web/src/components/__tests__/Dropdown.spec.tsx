import { createElement } from "react";
// import { render } from "@testing-library/react-native";
import { shallow } from "enzyme";
// import { mount, ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import DropdownCom, { DropdownProps } from "../Dropdown";
// import { BreadcrumbWebContainerProps } from "../../typings/BreadcrumbWebProps";

// const defaultProps: BreadcrumbWebContainerProps = {
//     name: "BreadcrumbWebTest",
//     class: "",
//     staticData: [],
//     isJump: false,
//     separator: "-"
// };

describe("BreadcrumbWeb", () => {
    let defaultProps: DropdownProps;
    beforeEach(() => {
        defaultProps = {
            trigger: "hover",
            placement: "bottomCenter",
            arrow: false,
            overlay: jest.fn(),
            children: null
        };
    });

    it("renders default dropdown", () => {
        const dropdown = shallow(<DropdownCom {...defaultProps} />);
        expect(dropdown).toMatchSnapshot();
    });

    it("see the arrow of dropdown", () => {
        const dropdown = shallow(<DropdownCom {...defaultProps} arrow />);
        expect(dropdown).toMatchSnapshot();
    });

    it("trigger click of dropdown", () => {
        const dropdown = shallow(<DropdownCom {...defaultProps} trigger="click" />);
        expect(dropdown).toMatchSnapshot();
    });

    it("placement topRight of dropdown", () => {
        const dropdown = shallow(<DropdownCom {...defaultProps} placement="topRight" />);
        expect(dropdown).toMatchSnapshot();
    });

    it("render custom children of dropdown", () => {
        const dropdown = shallow(
            <DropdownCom {...defaultProps}>
                <div>children</div>
            </DropdownCom>
        );
        expect(dropdown).toMatchSnapshot();
    });
});
