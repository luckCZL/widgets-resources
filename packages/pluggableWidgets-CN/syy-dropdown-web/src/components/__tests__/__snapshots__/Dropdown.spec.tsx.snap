// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWeb placement topRight of dropdown 1`] = `
<Dropdown
  arrow={false}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={[MockFunction]}
  placement="topRight"
  trigger={
    Array [
      "hover",
    ]
  }
/>
`;

exports[`BreadcrumbWeb render custom children of dropdown 1`] = `
<Dropdown
  arrow={false}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={[MockFunction]}
  placement="bottomCenter"
  trigger={
    Array [
      "hover",
    ]
  }
>
  <div>
    children
  </div>
</Dropdown>
`;

exports[`BreadcrumbWeb renders default dropdown 1`] = `
<Dropdown
  arrow={false}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={[MockFunction]}
  placement="bottomCenter"
  trigger={
    Array [
      "hover",
    ]
  }
/>
`;

exports[`BreadcrumbWeb see the arrow of dropdown 1`] = `
<Dropdown
  arrow={true}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={[MockFunction]}
  placement="bottomCenter"
  trigger={
    Array [
      "hover",
    ]
  }
/>
`;

exports[`BreadcrumbWeb trigger click of dropdown 1`] = `
<Dropdown
  arrow={false}
  mouseEnterDelay={0.15}
  mouseLeaveDelay={0.1}
  overlay={[MockFunction]}
  placement="bottomCenter"
  trigger={
    Array [
      "click",
    ]
  }
/>
`;
